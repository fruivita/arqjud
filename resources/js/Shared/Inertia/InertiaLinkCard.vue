<!--
    Link inertia estilizado como um card com Ã­cone e texto.

    @see https://vuejs.org/guide/introduction.html
    @see https://tailwindcss.com/docs
    @see https://inertiajs.com/
 -->

<script setup>
import Icone from '@/Shared/Icones/Icone.vue';
import { useStatusRequisicaoStore } from '@/Stores/StatusRequisicaoStore';

defineProps({
    icone: { type: String, required: true },
    texto: { type: String, required: true },
    href: { type: String, required: true },
});

const status = useStatusRequisicaoStore();
</script>

<template>
    <div
        class="rounded bg-primaria-300 shadow-lg shadow-secundaria-500 transition duration-300 hover:bg-primaria-200 focus:bg-primaria-200 dark:bg-secundaria-600 dark:shadow-primaria-500 hover:dark:bg-secundaria-500 focus:dark:bg-secundaria-500"
    >
        <InertiaLink
            :class="{
                'cursor-not-allowed opacity-50': status.processando,
            }"
            :href="href"
            class="flex flex-col items-center space-y-6 p-3"
        >
            <Icone :nome="icone" class="h-16 w-16" />

            <span class="break-words text-center">{{ texto }}</span>
        </InertiaLink>
    </div>
</template>
