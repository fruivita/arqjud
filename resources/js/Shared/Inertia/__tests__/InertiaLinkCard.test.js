/**
 * Testes para o componente InertiaLinkCard.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import InertiaLinkCard from '@/Shared/Inertia/InertiaLinkCard.vue';
import { createTestingPinia } from '@pinia/testing';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';

let mountFunction;
let pinia;

beforeEach(() => {
    pinia = (options = {}) =>
        createTestingPinia({
            initialState: {
                StatusRequisicaoStore: { ...options },
            },
        });

    mountFunction = (options = {}) => {
        return mount(InertiaLinkCard, { ...options });
    };
});

describe('InertiaLinkCard', () => {
    test('propriedades estão definidas', () => {
        expect(InertiaLinkCard.props).toMatchObject({
            icone: { type: String, required: true },
            href: { type: String, required: true },
            texto: { type: String, required: true },
        });
    });

    test('renderiza o componente respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { icone: 'key', href: 'http://foo.bar', texto: 'foo' },
                global: {
                    plugins: [pinia({ processando: false })],
                },
            }).html()
        ).toMatchSnapshot();
    });

    test('renderiza o componente de maneira diversa enquanto a página se carrega respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { icone: 'key', href: 'http://foo.bar', texto: 'foo' },
                global: {
                    plugins: [pinia({ processando: true })],
                },
            }).html()
        ).toMatchSnapshot();
    });
});
