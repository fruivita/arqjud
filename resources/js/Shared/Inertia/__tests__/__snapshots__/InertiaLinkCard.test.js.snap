// Vitest Snapshot v1

exports[`InertiaLinkCard > renderiza o componente de maneira diversa enquanto a p√°gina se carrega respeitando o snapshot 1`] = `
"<div class=\\"rounded bg-primaria-300 shadow-lg shadow-secundaria-500 transition duration-300 hover:bg-primaria-200 focus:bg-primaria-200 dark:bg-secundaria-600 dark:shadow-primaria-500 hover:dark:bg-secundaria-500 focus:dark:bg-secundaria-500\\">
  <inertialink class=\\"cursor-not-allowed opacity-50 flex flex-col items-center space-y-6 p-3\\" href=\\"http://foo.bar\\"><svg class=\\"h-16 w-16\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
      <path d=\\"M0 8a4 4 0 0 1 7.465-2H14a.5.5 0 0 1 .354.146l1.5 1.5a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0L13 9.207l-.646.647a.5.5 0 0 1-.708 0L11 9.207l-.646.647a.5.5 0 0 1-.708 0L9 9.207l-.646.647A.5.5 0 0 1 8 10h-.535A4 4 0 0 1 0 8zm4-3a3 3 0 1 0 2.712 4.285A.5.5 0 0 1 7.163 9h.63l.853-.854a.5.5 0 0 1 .708 0l.646.647.646-.647a.5.5 0 0 1 .708 0l.646.647.646-.647a.5.5 0 0 1 .708 0l.646.647.793-.793-1-1h-6.63a.5.5 0 0 1-.451-.285A3 3 0 0 0 4 5z\\"></path>
      <path d=\\"M4 8a1 1 0 1 1-2 0 1 1 0 0 1 2 0z\\"></path>
    </svg><span class=\\"break-words text-center\\">foo</span></inertialink>
</div>"
`;

exports[`InertiaLinkCard > renderiza o componente respeitando o snapshot 1`] = `
"<div class=\\"rounded bg-primaria-300 shadow-lg shadow-secundaria-500 transition duration-300 hover:bg-primaria-200 focus:bg-primaria-200 dark:bg-secundaria-600 dark:shadow-primaria-500 hover:dark:bg-secundaria-500 focus:dark:bg-secundaria-500\\">
  <inertialink class=\\"flex flex-col items-center space-y-6 p-3\\" href=\\"http://foo.bar\\"><svg class=\\"h-16 w-16\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
      <path d=\\"M0 8a4 4 0 0 1 7.465-2H14a.5.5 0 0 1 .354.146l1.5 1.5a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0L13 9.207l-.646.647a.5.5 0 0 1-.708 0L11 9.207l-.646.647a.5.5 0 0 1-.708 0L9 9.207l-.646.647A.5.5 0 0 1 8 10h-.535A4 4 0 0 1 0 8zm4-3a3 3 0 1 0 2.712 4.285A.5.5 0 0 1 7.163 9h.63l.853-.854a.5.5 0 0 1 .708 0l.646.647.646-.647a.5.5 0 0 1 .708 0l.646.647.646-.647a.5.5 0 0 1 .708 0l.646.647.793-.793-1-1h-6.63a.5.5 0 0 1-.451-.285A3 3 0 0 0 4 5z\\"></path>
      <path d=\\"M4 8a1 1 0 1 1-2 0 1 1 0 0 1 2 0z\\"></path>
    </svg><span class=\\"break-words text-center\\">foo</span></inertialink>
</div>"
`;
